File: code/community/1301688561354936322/arabic-language-support-deepgram-tts.py
import os
import requests

# Load the Deepgram API key from an environment variable
DEEPGRAM_API_KEY = os.getenv('DEEPGRAM_API_KEY')

url = "https://api.deepgram.com/v1/speak"

text_to_speak = "Hello, world!"

headers = {
    "Authorization": f"Bearer {DEEPGRAM_API_KEY}",
    "Content-Type": "application/json"
}

json_data = {
    "text": text_to_speak,
    "voice": "en-US"
}

response = requests.post(url, headers=headers, json=json_data)

if response.status_code == 200:
    print("Text-to-Speech executed successfully")
else:
    print("Failed to execute Text-to-Speech")

File: code/community/1301688561354936322/arabic-language-support-deepgram-tts.go
package main

import (
    "bytes"
    "encoding/json"
    "fmt"
    "net/http"
    "os"
)

func main() {
    apiKey := os.Getenv("DEEPGRAM_API_KEY")
    url := "https://api.deepgram.com/v1/speak"

    textToSpeak := "Hello, world!"

    requestBody, err := json.Marshal(map[string]string{
        "text": textToSpeak,
        "voice": "en-US",
    })
    if err != nil {
        panic(err)
    }

    client := &http.Client{}
    request, err := http.NewRequest("POST", url, bytes.NewBuffer(requestBody))
    if err != nil {
        panic(err)
    }

    request.Header.Set("Authorization", fmt.Sprintf("Bearer %s", apiKey))
    request.Header.Set("Content-Type", "application/json")

    response, err := client.Do(request)
    if err != nil {
        panic(err)
    }
    defer response.Body.Close()

    if response.StatusCode == http.StatusOK {
        fmt.Println("Text-to-Speech executed successfully")
    } else {
        fmt.Println("Failed to execute Text-to-Speech")
    }
}